codegenerator/cli/npm/envio/src/ReorgDetection.res

```diff
@@ -121,13 +133,13 @@ module LastBlockScannedHashes: {
     ) {
     | Some(scannedBlock) if scannedBlock.blockHash !== lastBlockScannedData.blockHash =>
       Some({
-        reorgGuard,
+        receivedBlock: lastBlockScannedData,
         scannedBlock,
       })
     | _ =>
       switch firstBlockParentNumberAndHash {
-      //If parentHash is None, either it's the genesis block (no reorg)
-      //Or its already confirmed so no Reorg
+      //If parentHash is None, then it's the genesis block (no reorg)
+      //Need to check that parentHash matches because of the dynamic contracts
```

Comment:
Not sure of this comment 🤔 I think it's the last block scanned we need to check due to dynamic contract partitions merging right?

Reply:
True

Reply:
True